deps/tools/jlchecksum

```diff
@@ -21,32 +23,50 @@ print_hash()
     if [ ${#1} -gt 64 ]; then
         NUM_LINES=$(( (${#1} + 63) / 64))
         for i in `seq 0 1 $((NUM_LINES - 1))`; do
-            str_piece=$(echo "$1" | awk "{ string=substr(\$0, $((i*64 + 1)), $(((i+1)*64))); print string; }")
+            str_piece=$(echo "$1" | awk "{ string=substr(\$0, $((i*64 + 1)), 64); print string; }")
             echo "      $str_piece"
         done
     else
         echo "      $1" >&2
     fi
 }
 
+abspath()
+{
+    echo $(cd $(dirname "$1"); pwd)/$BASENAME;
+}
+
 checksum_error()
 {
+    ARG1=$(realpath "$ARG1" 2>/dev/null || echo $(abspath "$ARG1"))
     echo "===============================================================================" >&2
     echo "  ERROR: $CHECKSUM_TYPE checksum failure on $BASENAME, should be:" >&2
     print_hash "$TRUE_CHECKSUM"
     echo "  But \`$CHECKSUM_PROG\` results in:" >&2
     print_hash "$CURR_CHECKSUM"
     echo "  This can happen due to bad downloads or network proxies, please check your" >&2
     echo "  network proxy/firewall settings and delete" >&2
-    echo "  $(realpath $ARG1)" >&2
+    echo "  $ARG1" >&2
     echo "  to force a redownload when you are ready" >&2
     echo "===============================================================================" >&2
     exit 2
 }
 
 find_checksum()
 {
+    for pack in "$DEPSDIR"/checksums/*; do
+        if [ -f "$pack" ]; then
+            TRUE_CHECKSUM=$(awk -F / "{ if (\$1 == \"$BASENAME\" && \$2 == \"$CHECKSUM_TYPE\") print \$3 }" "$pack")
+            if [ $TRUE_CHECKSUM ]; then
+                return
+            fi
+        fi
+    done
```

Comment:
I bet we can speed this up significantly via `grep $BASENAME/$CHECKSUM_TYPE $DEPSDIR/checksums/* | cut -d'/' -f 3`

Reply:
I tried that, but gnu grep doesn't support `\Q` and I don't want to escape it with sed. Also, it's just 30 files.